File: inorder_treetraversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> mylist =new ArrayList<Integer>();
        Stack<TreeNode> stack = new Stack<TreeNode>();
        TreeNode node = root;
        while(!stack.isEmpty() || node!=null){
            if(node !=null){
                stack.push(node);
                node = node.left;
            }
            else{
                node = stack.pop();
                mylist.add(node.val);
                node = node.right;
            }
        }
        return mylist;
    }
}

File: inorder_tree_traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:#notice pythonic condition style
    # @param root, a tree node
    # @return a list of integers
    def inorderTraversal(self, root):
        stack = []
        result = []
        node = root
        while node or stack:
            if node:
                stack.append(node)
                node= node.left
            else:
                node = stack.pop()
                result.append(node.val)
                node = node.right
        return result